### PR TITLE
New IaC Policies for July 2024

### PR DESCRIPTION
**Pull request by tsmithv11**

### CODE REVIEW

**Comment by tsmithv11** [comment]
TODO Post review:

1. Add to summaries
2. Add to book
3. Add policy IDs

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+|Build
+
+|Frameworks
+|Terraform,TerraformPlan,ARM,Bicep
+
+|===
+
+=== Description
+
+This policy ensures that isolated compute is enabled for Synapse Spark pools in Azure. Isolated compute involves using dedicated resources that are not shared with other customers. This enhances security by minimizing the risk of data breaches or unauthorized access, as the compute resources are isolated to your environment. Enabling isolated compute is crucial for maintaining high security standards, especially when handling sensitive data or running critical applications in Synapse Spark pools.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc:28]: This policy detects whether isolated compute is disabled for Synapse Spark pools in Azure. Isolated compute utilizes dedicated resources that are not shared with other customers. This isolation minimizes data breach and unauthorized access risks, enhancing security for your Spark pools. Enabling isolated compute is crucial for maintaining high security standards, especially when handling sensitive data or running critical applications within Synapse Spark pools.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+=== Fix - Buildtime
+
+*Terraform*
+
+* *Resource:* azurerm_synapse_spark_pool
+* *Arguments:* compute_isolation_enabled
+
+In order to fix this issue, ensure that `compute_isolation_enabled` is set to `true`.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc:37]: Move to line 31 and see comments there.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+|Terraform,TerraformPlan,ARM,Bicep
+
+|===
+
+=== Description
+
+This policy ensures that isolated compute is enabled for Synapse Spark pools in Azure. Isolated compute involves using dedicated resources that are not shared with other customers. This enhances security by minimizing the risk of data breaches or unauthorized access, as the compute resources are isolated to your environment. Enabling isolated compute is crucial for maintaining high security standards, especially when handling sensitive data or running critical applications in Synapse Spark pools.
+
+=== Fix - Buildtime
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc:31]: - Isolate your workloads from other customers by enabling isolate compute in your Azure Synapse Spark pool: Set the `compute_isolation_enabled` property to `true` in the pool's configuration.
- Enable isolated compute in your Azure Synapse Spark pool: Set the `compute_isolation_enabled` property to `true` in the pool's configuration.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+}
+----
+
+
+*ARM*
+
+* *Resource:* Microsoft.Synapse/workspaces/bigDataPools
+* *Arguments:* properties/isComputeIsolationEnabled
+
+In order to fix this issue, ensure that `properties/isComputeIsolationEnabled` is set to `true`.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-242.adoc:53]: In this example, the `properties/isComputeIsolationEnabled` property is set to `true` within the properties section of the Microsoft.Synapse/workspaces/bigDataPools resource definition.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc [hunk 5]
```diff
@@ -0,0 +1,50 @@
+
+== Azure Synapse Workspace does not have extended audit logs
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc:2]: - Missing extended audit logs for Azure Synapse Workspace
- No extended audit logs for Azure Synapse Workspace

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+|Build
+
+|Frameworks
+|Terraform,TerraformPlan
+
+|===
+
+=== Description
+
+This policy ensures that an Azure Synapse Workspace has extended audit logs enabled. Extended audit logs are important because they provide detailed records of access and activity within the workspace. Without these logs, it can be challenging to monitor and investigate any unauthorized access or anomalies, thus posing a security risk. Logging mechanisms are essential for tracking user actions, policy breaches, data access, and troubleshooting issues, making it a critical practice for maintaining security and compliance within Azure environments.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc:28]: - This policy detects whether extended audit logs are disabled on an Azure Synapse Workspace. Logging mechanisms are essential for tracking user actions, policy breaches, data access, and troubleshooting issues, making it a critical practice for maintaining security and compliance within Azure environments.
Extended audit logs provide detailed records of access and activity within the workspace, which are crucial for monitoring and investigating potential security risks like unauthorized access or anomalies.
- This policy detects whether extended audit logs are disabled on an Azure Synapse Workspace. Extended audit logs provide detailed records of access and activity within the workspace, which are crucial for monitoring and investigating potential security risks such as unauthorized access or anomalies. 
Logging mechanisms are essential for tracking user actions, policy breaches, data access, and troubleshooting issues, making it a critical practice for maintaining security and compliance within Azure environments.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+=== Fix - Buildtime
+
+*Terraform*
+
+* *Resource:* azurerm_synapse_workspace
+* *Arguments:* azurerm_synapse_workspace_extended_auditing_policy
+
+In order to fix this issue, ensure that all `azurerm_synapse_workspace` resources are connected to a `azurerm_synapse_workspace_extended_auditing_policy`.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc:37]: Move to line 31 - see comments there

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+|Terraform,TerraformPlan
+
+|===
+
+=== Description
+
+This policy ensures that an Azure Synapse Workspace has extended audit logs enabled. Extended audit logs are important because they provide detailed records of access and activity within the workspace. Without these logs, it can be challenging to monitor and investigate any unauthorized access or anomalies, thus posing a security risk. Logging mechanisms are essential for tracking user actions, policy breaches, data access, and troubleshooting issues, making it a critical practice for maintaining security and compliance within Azure environments.
+
+=== Fix - Buildtime
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc:31]: Ensure that all `azurerm_synapse_workspace` resources are connected to a `azurerm_synapse_workspace_extended_auditing_policy`.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+*Terraform*
+
+* *Resource:* azurerm_synapse_workspace
+* *Arguments:* azurerm_synapse_workspace_extended_auditing_policy
+
+In order to fix this issue, ensure that all `azurerm_synapse_workspace` resources are connected to a `azurerm_synapse_workspace_extended_auditing_policy`.
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-53.adoc:38]: In this example, extended audit logging is configured for your Azure Synapse Workspace using Azure Resource Manager (ARM) templates.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc [hunk 10]
```diff
@@ -0,0 +1,51 @@
+
+== Azure Synapse SQL Pool does not have log monitoring enabled
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc:2]: Log monitoring disabled for Azure Synapse SQL Pool

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+|Build
+
+|Frameworks
+|Terraform,TerraformPlan
+
+|===
+
+=== Description
+
+This policy is verifying whether log monitoring is active for Synapse SQL Pools in Microsoft Azure. This is an important security measure as it helps in tracking database activities, detecting suspicious behavior, and responding to potential security incidents. Without log monitoring, organizations may be unable to detect unauthorized access or misuse of data, making it harder to protect sensitive information and meet compliance requirements.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc:28]: This policy detects whether log monitoring is enabled for Synapse SQL Pools in Microsoft Azure. Log monitoring is an important security measure as it helps in tracking database activities, detecting suspicious behavior, and responding to potential security incidents. Without log monitoring, organizations may be unable to detect unauthorized access or misuse of data, making it harder to protect sensitive information and meet compliance requirements.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+=== Fix - Buildtime
+
+*Terraform*
+
+* *Resource:* azurerm_synapse_sql_pool, azurerm_synapse_sql_pool_extended_auditing_policy
+* *Arguments:* log_monitoring_enabled
+
+In order to fix this issue, ensure that all `azurerm_synapse_sql_pool` are connected to `azurerm_synapse_sql_pool_extended_auditing_policy` and that `log_monitoring_enabled` is set to `true`.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc:37]: Move to line 31 - see comments there

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+|Terraform,TerraformPlan
+
+|===
+
+=== Description
+
+This policy is verifying whether log monitoring is active for Synapse SQL Pools in Microsoft Azure. This is an important security measure as it helps in tracking database activities, detecting suspicious behavior, and responding to potential security incidents. Without log monitoring, organizations may be unable to detect unauthorized access or misuse of data, making it harder to protect sensitive information and meet compliance requirements.
+
+=== Fix - Buildtime
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc:31]: 1. Ensure all `azurerm_synapse_sql_pool` resources are associated with an `azurerm_synapse_sql_pool_extended_auditing_policy` resource.
2. Set the `log_monitoring_enabled property` to `true` in each `azurerm_synapse_sql_pool_extended_auditing_policy` resource.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+*Terraform*
+
+* *Resource:* azurerm_synapse_sql_pool, azurerm_synapse_sql_pool_extended_auditing_policy
+* *Arguments:* log_monitoring_enabled
+
+In order to fix this issue, ensure that all `azurerm_synapse_sql_pool` are connected to `azurerm_synapse_sql_pool_extended_auditing_policy` and that `log_monitoring_enabled` is set to `true`.
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azure-2-54.adoc:38]: In this example, log monitoring for Azure Synapse SQL Pools is enforced using Azure Resource Manager (ARM) templates.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-2-51.adoc [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+=== Fix - Buildtime
+
+*Terraform*
+
+* *Resource:* azurerm_synapse_sql_pool
+* *Arguments:* azurerm_synapse_sql_pool_security_alert_policy
+
+To fix this issue, you must configure a security alert policy for your Synapse SQL Pool. Any `azurerm_synapse_sql_pool` must be connected to a `azurerm_synapse_sql_pool_security_alert_policy` block that is used to define the security alert policies.
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-2-51.adoc:37]: In this example, a security alert policy for an Azure Synapse SQL Pool is configured using Azure Resource Manager (ARM) templates.

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-2-52.adoc [hunk 16]
```diff
@@ -0,0 +1,49 @@
+
+== Azure Synapse SQL Pool does not have vulnerability assessment enabled
```
[the remark this answers is not in the document]
Comment (tsmithv11) [reply docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-2-52.adoc:2]: First option is great. Thanks!

File: docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-2-52.adoc [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+The policy is looking to ensure that a Synapse SQL Pool in an Azure environment has a vulnerability assessment attached. This is important because a vulnerability assessment helps identify and mitigate potential security risks by detecting vulnerabilities and providing recommendations to address them. Not having such an assessment in place can leave your SQL Pool exposed to potential security threats, which could lead to data breaches, loss, or other types of cyberattacks. Monitoring and maintaining the security posture of your database services is crucial for protecting sensitive data and complying with regulatory standards.
+
+=== Fix - Buildtime
+
+*Terraform*
+
+* *Resource:* azurerm_synapse_sql_pool
+* *Arguments:* azurerm_synapse_sql_pool_security_alert_policy, recurring_scans.*.enabled
+
+In order to fix this issue, ensure that all `azurerm_synapse_sql_pool` are linked to a `azurerm_synapse_sql_pool_security_alert_policy` with `recurring_scans` enabled.
```
[the remark this answers is not in the document]
Comment (tsmithv11) [reply docs/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azure-2-52.adoc:37]: It has to be below the Terraform title to be picked up by our automation

**Pull request merged**